code/game/objects/objs.dm

```diff
@@ -32,6 +34,9 @@
 	else if (!istype(armor, /datum/armor))
 		stack_trace("Invalid type [armor.type] found in .armor during /obj Initialize()")
 
+	if(obj_integrity == null)
```

Comment:
```suggestion
	if(isnull(obj_integrity))
```

Reply:
Else it will return true if `obj_integrity` is zero.

Reply:
no thats not how this works.
What happens is `obj_integrity` is defined as null, then in this `Initialize` if a child type hasnt defined some value it just sets to the max.  Things shouldnt be defined with a value of 0.

Reply:
https://github.com/tgstation/tgstation/blob/01d9c7d9a9d0f6cc5bf562231e75671a63564e15/code/game/objects/objs.dm#L54

Reply:
```
var/a = 0
world.log << "isnull(a): [isnull(a)]"
world.log << "a == null: [a == 0]"
world.log << "!a: [!a]"
```
Results:
```
isnull(a): 0
a == null: 1
!a: 1
```
I'm aware `obj_integrity` is defined as `null`. I'd just rather avoid `if(x == null)` because it's deceptive when it's a numerical variable.

Reply:
Actually scratch that. I made a typo while testing, and wrongly got that `x == null` would return `1` if `x` was `0`.